src/scitokens_internal.h

```diff
@@ -418,24 +471,78 @@ class Validator {
     }
 
     void verify(const SciToken &scitoken, time_t expiry_time) {
-        auto result = verify_async(scitoken);
-        while (!result->m_done) {
-            auto timeout_val = result->get_timeout_val(expiry_time);
-            select(result->get_max_fd() + 1, result->get_read_fd_set(),
-                   result->get_write_fd_set(), result->get_exc_fd_set(),
-                   &timeout_val);
-            if (time(NULL) >= expiry_time) {
-                throw CurlException("Timeout when loading the OIDC metadata.");
+        std::string issuer = "";
+        auto start_time = std::chrono::steady_clock::now();
+        bool has_issuer = false;
+        
+        try {
+            // Try to extract issuer for monitoring
+            if (scitoken.m_decoded && scitoken.m_decoded->has_payload_claim("iss")) {
+                issuer = scitoken.m_decoded->get_issuer();
+                has_issuer = true;
             }
+            
+            auto result = verify_async(scitoken);
+            while (!result->m_done) {
+                auto timeout_val = result->get_timeout_val(expiry_time);
+                select(result->get_max_fd() + 1, result->get_read_fd_set(),
+                       result->get_write_fd_set(), result->get_exc_fd_set(),
+                       &timeout_val);
+                if (time(NULL) >= expiry_time) {
+                    throw CurlException("Timeout when loading the OIDC metadata.");
+                }
 
-            result = verify_async_continue(std::move(result));
+                result = verify_async_continue(std::move(result));
+            }
+            
+            // Record successful validation
+            if (has_issuer) {
+                auto end_time = std::chrono::steady_clock::now();
+                auto duration = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                    end_time - start_time);
+                internal::MonitoringStats::instance().record_validation_success(
```

Comment:
Since the issuer is unchanged throughout the call, we should be able to get a handle to the issuer stats and update it directly.  That avoids needing multiple locks.

Additionally, this only updates when the operation finished - which can take significant time.  Do periodic updates of the total duration, having the select block for no more than 50ms at a time.